### PR TITLE
feat(tiflow): upload tiflow ut codecoverage if test passed

### PR DESCRIPTION
**Pull request by purelind**
* upload tiflow ut codecoverage if test passed
* upload code coveage in master and lts branch's pull requests

### CODE REVIEW

**Comment by purelind** [comment]
repaly tests
- [x] master https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftiflow%2Fghpr_verify/detail/ghpr_verify/185/pipeline/158
- [ ] release-7.5 
- [ ] release-7.1
- [ ] release-6.5
- [ ] release-6.1

**Comment by purelind** [comment]
we need to remove the upload step in the tiflow makefile. https://github.com/pingcap/tiflow/pull/10446

File: pipelines/pingcap/tiflow/latest/ghpr_verify.groovy [hunk 1]
```diff
@@ -85,13 +84,28 @@ pipeline {
                             }
                         }
                         post {
+                            success {
+                                dir('tiflow') {
+                                    script {
+                                        def testConfigs = [
+                                            dm_unit_test_in_verify_ci: [flags: "unit,dm", coverage_file: "/tmp/dm_test/cov.unit_test.out"],
+                                            unit_test_in_verify_ci: [flags: "unit,cdc", coverage_file: "/tmp/tidb_cdc_test/cov.unit.out"],
+                                            engine_unit_test_in_verify_ci: [flags: "unit,engine", coverage_file: "/tmp/engine_test/cov.unit_test.out"]
```
Comment (wuhuizuo) [inline pipelines/pingcap/tiflow/latest/ghpr_verify.groovy:93]: Ref: 
- https://github.com/PingCAP-QE/ci/issues/2171


Let's migrate to use flags to separate test types. components will be identified auto by file path defined in repo's `codecov.yaml` file.

**Comment by wuhuizuo** [comment]
> * upload tiflow ut codecoverage if test passed
> * upload code coveage in master and lts branch

But it will just upload coverage for pull request, we also need another postsubmit job to upload coverage to base branch

File: pipelines/pingcap/tiflow/latest/ghpr_verify.groovy (hunk 1 above)
Comment (purelind) [reply pipelines/pingcap/tiflow/latest/ghpr_verify.groovy:93]: Does this PR https://github.com/pingcap/tiflow/pull/9252 need to be cherry-picked to other LTS branches?

**Review by wuhuizuo — APPROVED** [review]

**Pull request merged**